java/adapter/jdbc/src/main/java/org/apache/arrow/adapter/jdbc/JdbcToArrowConfig.java

```diff
@@ -102,9 +105,10 @@ public final class JdbcToArrowConfig {
           Map<Integer, JdbcFieldInfo> arraySubTypesByColumnIndex,
           Map<String, JdbcFieldInfo> arraySubTypesByColumnName,
           int targetBatchSize,
-          Function<JdbcFieldInfo, ArrowType> jdbcToArrowTypeConverter) {
+          Function<JdbcFieldInfo, ArrowType> jdbcToArrowTypeConverter,
+          Function4Arity<ArrowType, Integer, Boolean, FieldVector, JdbcConsumer> jdbcConsumerGetter) {
```

Comment:
Could we add a delegate constructor to avoid breaking existing code?

Reply:
done!